synapse/storage/databases/main/client_ips.py

```diff
@@ -410,8 +410,8 @@ def _prune_old_user_ips_txn(txn):
 class ClientIpStore(ClientIpWorkerStore):
     def __init__(self, database: DatabasePool, db_conn, hs):
 
-        self.client_ip_last_seen = DeferredCache(
-            name="client_ip_last_seen", keylen=4, max_entries=50000
+        self.client_ip_last_seen = LruCache(
+            cache_name="client_ip_last_seen", keylen=4, max_size=50000
```

Comment:
I don't think giving a `keylen` matters here since it isn't used by default? The 4 also seems wrong, since `key = (user_id, access_token, ip)`.

I think this comment is true of all of the caches touched in this PR, actually.

Reply:
yeahh... would you prefer I remove them?

Reply:
It isn't really related to this, I just noticed it here, we can do it in a follow-up or...something else? I don't know if you have plans for dealing with `TreeCache` at all.

Reply:
not really. I was hoping to avoid digging down into that layer.

Reply:
Bottom line: I don't think you need to fix it here.